tools/iree-e2e-matmul-test.c

```diff
@@ -656,12 +662,27 @@ static bool matmul_result_elements_agree(iree_e2e_test_value_t expected,
     case IREE_E2E_TEST_VALUE_TYPE_I32:
       return actual.i32 == expected.i32;
     // Since we fill buffers with small integers for floating point GEMMs
-    // functional testing, we test for bit-exactness on the actual and
-    // expected values.
+    // functional testing, we can test for bit-exactness on the actual and
+    // expected values. Inexact results are only permitted when the
+    // `require_exact_results` flag is set to `false`.
     case IREE_E2E_TEST_VALUE_TYPE_F16:
-      return actual.f16_u16 == expected.f16_u16;
+      if (actual.f16_u16 == expected.f16_u16) {
```

Comment:
Nit: we can drop these `{..}` to make them onliner?

Reply:
Sure. I typically keep `return` on a separate line to make adding breakpoints easier, but in this case it's unlikely to be helpful anyway.